EventListener/UserSubscriber.php

```diff
@@ -44,9 +50,43 @@ public static function getSubscribedEvents()
     {
         return array(
             UserEvents::USER_FORM_AUTHENTICATION => array('onUserFormAuthentication', 0),
+            UserEvents::USER_PRE_AUTHENTICATION  => array('onUserSsoAuthentication', 0),
         );
     }
+    /**
+     * Authenticate via the form using users defined in LDAP server(s).
+     *
+     * @param AuthenticationEvent $event
+     *
+     * @return bool|void
+     */
+    public function onUserSsoAuthentication(AuthenticationEvent $event)
+    {
+        $username = $_SERVER['PHP_AUTH_USER'];
+        $password = $_SERVER['PHP_AUTH_PW'];
```

Comment:
Get those directly from `$_SERVER` is not very secure. I do not remember how it's done but I'm pretty sure there is a better / more secure way to access those from Symfony.